front/src/Stores/BiggestAvailableAreaStore.ts

```diff
@@ -1,112 +1,86 @@
-
-        const presentationArea =
-            (game.offsetHeight - (lastPresentationDiv.offsetTop + lastPresentationDiv.offsetHeight)) *
-            (lastPresentationDiv.offsetLeft + lastPresentationDiv.offsetWidth);
-
-        let leftSideBar: number;
-        let bottomSideBar: number;
-        if (sidebarChildren.length === 0) {
-            leftSideBar = HtmlUtils.getElementByIdOrFail<HTMLDivElement>("sidebar").offsetLeft;
-            bottomSideBar = 0;
-        } else {
-            const lastSideBarChildren = sidebarChildren[sidebarChildren.length - 1];
-            leftSideBar = lastSideBarChildren.offsetLeft;
-            bottomSideBar = lastSideBarChildren.offsetTop + lastSideBarChildren.offsetHeight;
+    // populate with all vertices
+    for (const blocker of blockers) {
+        verticesX.push(blocker.xStart, blocker.xEnd);
+        verticesY.push(blocker.yStart, blocker.yEnd);
+    }
+    verticesX.sort();
+    verticesY.sort();
+
+    // NOTE: cannot use fill() here, it makes references to a single array
+    const occupiedGrid: boolean[][] = new Array(verticesY.length - 1);
+    for (let j = 0; j < verticesY.length - 1; j += 1) {
+        occupiedGrid[j] = new Array(verticesX.length - 1).fill(false);
+    }
+
+    for (const blocker of blockers) {
+        const [left, top] = getGridCoordinates(blocker.xStart, blocker.yStart, verticesX, verticesY);
+        const [right, bottom] = getGridCoordinates(blocker.xEnd, blocker.yEnd, verticesX, verticesY);
+        for (let j = top; j < bottom; j += 1) {
+            for (let i = left; i < right; i += 1) {
+                occupiedGrid[j][i] = true;
+            }
         }
-        const sideBarArea = (game.offsetWidth - leftSideBar) * (game.offsetHeight - bottomSideBar);
-
-        if (presentationArea <= sideBarArea) {
-            return {
-                xStart: leftSideBar,
-                yStart: bottomSideBar,
-                xEnd: game.offsetWidth,
-                yEnd: game.offsetHeight,
-            };
-        } else {
-            return {
-                xStart: 0,
-                yStart: lastPresentationDiv.offsetTop + lastPresentationDiv.offsetHeight,
-                xEnd: /*lastPresentationDiv.offsetLeft + lastPresentationDiv.offsetWidth*/ game.offsetWidth, // To avoid flickering when a chat start, we center on the center of the screen, not the center of the main content area
-                yEnd: game.offsetHeight,
-            };
+    }
+
+    // create an array of all free boxes
+    const freeBoxes: Box[] = [];
+    for (let x = 0; x < occupiedGrid.length - 1; x += 1) {
+        for (let y = 0; y < occupiedGrid[x].length - 1; y += 1) {
+            freeBoxes.push(...findAllFreeBoxes(x, y, occupiedGrid));
         }
     }
+
+    const biggestFreeArea = freeBoxes
+        .map((box) => ({ box, area: getBoxArea(box, verticesX, verticesY) }))
+        .sort((a, b) => b.area - a.area)[0];
```

Comment:
Micro optimization:

You can probably avoid the "sort" here and do a simple scan to find the biggest area.

The sort will be done in `O(n*log(n))` while simply scanning to find the biggest value can be done in `O(n)`